### PR TITLE
Fix cucumber tests for Windows use

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
Windows port numbers higher than 49999 found to be a problem, specifically the static assignment of port 50071 for the merge mining proxy server. This fix allows selection of free port numbers in the range 19000 to 25000 for all instances of GRPC servers, using existing `getFreePort` function. This also allows using the merge mining proxy in consecutive tests as the port is not freed up after a single test for multiple test usage.

## Motivation and Context
See **Description**

## How Has This Been Tested?
Tested running the cucumber tests in Windows.

## Types of changes

* [X] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch.
* [ ] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: integration_tests/helpers/mergeMiningProxyProcess.js [hunk 1]
```diff
@@ -15,12 +15,11 @@ class MergeMiningProxyProcess {
...
     async init() {
-        this.port = await getFreePort(19000, 20000);
+        this.port = await getFreePort(19000, 25000);
         this.name = `MMProxy${this.port}-${this.name}`;
         this.baseDir = `./temp/base_nodes/${dateFormat(new Date(), "yyyymmddHHMM")}/${this.name}`;
-    //    await this.run("cargo",
-//
-  //          ["run", "--release", "--bin", "tari_merge_mining_proxy", "--", "--base-path", ".", "--init"]);
+        this.grpcPort = await getFreePort(19000, 25000);
+        //console.log("MergeMiningProxyProcess init - assign server GRPC:", this.grpcPort);
```
Comment (StriderDM) [inline integration_tests/helpers/mergeMiningProxyProcess.js:22]: Extra comment left in perhaps?

File: integration_tests/helpers/mergeMiningProxyProcess.js [hunk 2]
```diff
@@ -121,8 +119,9 @@ class MergeMiningProxyProcess {
     }
 
     createClient() {
-        // Mostly only going to need one proxy
-        return new MergeMiningProxyClient("http://127.0.0.1:50071");
+        let address = "http://127.0.0.1:" + this.grpcPort;
+        //console.log("MergeMiningProxyProcess createClient - client address:", address);
```
Comment (StriderDM) [inline integration_tests/helpers/mergeMiningProxyProcess.js:123]: Extra comment left in perhaps?

**Review by StriderDM — APPROVED** [review]
Looks good. Appears that it should work.

**Review by stringhandler — APPROVED** [review]
Seems ok. Will have to test locally

**Pull request merged**